packages/dev/core/src/Engines/WebGL/webGLPipelineContext.ts

```diff
@@ -23,6 +40,49 @@ export class WebGLPipelineContext implements IPipelineContext {
     public programLinkError: Nullable<string> = null;
     public programValidationError: Nullable<string> = null;
 
+    constructor() {
+        const args: any[] = [];
+        const proxyFunction: (functionName: string) => ((/*uniformName: string, ...payload: any[]*/) => void) | undefined = (functionName: string) => {
+            const cacheFunction = cacheToSetProxyReference[functionName as string];
+            if (cacheFunction) {
+                const cacheFunc = this[`_cache${cacheFunction}` as Partial<keyof WebGLPipelineContext>];
+                return function (this: WebGLPipelineContext /*uniformName: string, ...payload: any[]*/) {
+                    const func = this.engine[functionName as keyof ThinEngine];

```

Comment:
this could be moved before the return

Reply:
No, it can't,m because engine is assigned after the function is defined.

Reply:
so should we move this in engine assignment ? or is it overkill ?

Reply:
a bit of overkill for one line that will be ca. 10 characters. We will need to add a setter and getter for engine, add the rest to be members of the class, this will add more code than it saves.